### PR TITLE
examples/ucontext-cp: more complex example to show how ucontext works with io_uring

### PR DESCRIPTION
**Pull request by CarterLi**
also add add timeout to indicate that different coroutines run in parallel actually

```shell
$ ./ucontext-cp io_uring-cp.c io_uring-cp.c.1 io_uring-test.c io_uring-test.c1 link-cp.c link-cp.c1
copying 3 files...
4->5: readv 1024 bytes from 0
6->7: readv 1024 bytes from 0
8->9: readv 1024 bytes from 0
4->5: writev 1024 bytes from 0
6->7: writev 1024 bytes from 0
8->9: writev 1024 bytes from 0
4->5: wait 1 second(s)
8->9: wait 1 second(s)
6->7: wait 1 second(s)
4->5: readv 1024 bytes from 1024
8->9: readv 1024 bytes from 1024
6->7: readv 1024 bytes from 1024
4->5: writev 1024 bytes from 1024
8->9: writev 1024 bytes from 1024
6->7: writev 785 bytes from 1024
4->5: wait 1 second(s)
8->9: wait 1 second(s)
6->7: done with ret code 0
4->5: readv 1024 bytes from 2048
8->9: readv 1024 bytes from 2048
4->5: writev 1024 bytes from 2048
8->9: writev 1024 bytes from 2048
4->5: wait 1 second(s)
8->9: wait 1 second(s)
4->5: readv 1024 bytes from 3072
8->9: readv 1024 bytes from 3072
4->5: writev 1024 bytes from 3072
8->9: writev 428 bytes from 3072
4->5: wait 1 second(s)
8->9: done with ret code 0
4->5: readv 1024 bytes from 4096
4->5: writev 745 bytes from 4096
4->5: done with ret code 0
finished with 3 success(es) and 0 failure(s)
```

The order of printed lines may vary for every execution

### CODE REVIEW

File: examples/ucontext-cp.c [hunk 1]
```diff
@@ -60,23 +62,18 @@ DEFINE_AWAIT_OP(readv)
...
-
+	pctx->ring = ring;
 	ret = getcontext(&pctx->ctx_fnew);
 	if (ret < 0) {
 		perror("getcontext");
 		return -1;
 	}
 	pctx->ctx_fnew.uc_stack.ss_sp = &pctx->stack_buf;
-	pctx->ctx_fnew.uc_stack.ss_size = sizeof(pctx->stack_buf);
+	pctx->ctx_fnew.uc_stack.ss_size = sizeof pctx->stack_buf;
```
Comment (axboe) [inline examples/ucontext-cp.c]: Please don't do that, it's correct as it is. sizeof() should include parens. Same in a few other spots.

**Comment by CarterLi** [comment]
done @axboe 

And suggestion: `io_uring_submit_and_wait` should always wait as its name reads `io_uring_submit_and_wait` instead of `io_uring_submit_and_wait_if_any_sqes_submitted`

**Comment by axboe** [comment]
> And suggestion: io_uring_submit_and_wait should always wait as its name reads io_uring_submit_and_wait instead of io_uring_submit_and_wait_if_any_sqes_submitted

Well, arguably it would be a bug to call that function if you don't have pending IO. That's the intent of it, submit IO and wait for at least one of them.

**Comment by axboe** [comment]
Actually, I take that back, it should always wait for an event. And looking at it, it should do that. What am I missing?

**Comment by axboe** [comment]
Oh my bad, looking at __io_uring_get_cqe(), not __io_uring_submit() with wait != 0.

**Pull request merged**